prov/efa/src/rxr/rxr_msg.c

```diff
@@ -786,6 +787,55 @@ struct rxr_rx_entry *rxr_msg_split_rx_entry(struct rxr_ep *ep,
 	return rx_entry;
 }
 
+/**
+ * @brief find an unexpected rx entry for a receive operation.
+ *
+ * @param[in]	ep	endpoint
+ * @param[in]	addr	fi_addr of the peer want to receive from, can be FI_ADDR_UNSPEC
+ * @param[in]	tag	tag of the unexpected message, used only if op is ofi_op_tagged.
+ * @param[in]	ignore	mask of the tag, used only if op is ofi_op_tagged.
+ * @param[in]	op	either ofi_op_tagged or ofi_op_msg.
+ * @return	If an unexpected rx_entry was found, return the pointer.
+ * 		Otherwise, return NULL.
+ */
+static inline
+struct rxr_rx_entry *rxr_msg_find_unexp_rx_entry(struct rxr_ep *ep, fi_addr_t addr,
+						 int64_t tag, uint64_t ignore, uint32_t op)
+{
+	struct rxr_match_info match_info;
+	struct dlist_entry *match;
+	struct rdm_peer *peer;
+
+	peer = (ep->util_ep.caps & FI_DIRECTED_RECV) ? rxr_ep_get_peer(ep, addr) : NULL;
+
+	if (op == ofi_op_msg) {
```

Comment:
op can only be one of ofi_op_msg or ofi_op_tagged, as you say.  This code would be a little easier to understand if you organized it as:

```
switch (op) {
case ofi_op_msg:
case ofi_op_msg:

default:
  ERROR("Illegal op value %d\n", op);
  abort();
}
```

It takes significantly less cognitive load on the reader and has no change in runtime (may actually be faster, as the compiler should emit a jump table rather than a conditional).

Reply:
Thanks! fixed in new revision.